ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -499,67 +368,24 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
       }
     }
 
-    if (computeStats) {
```

Comment:
that should stay here

Reply:
I moved `computeStats` it to CompactionExecutor and returned the logic to update stats at the end of `findNextCompactionAndExecute` as following:

```
    if (Optional.ofNullable(compactionExecutor).map(CompactionExecutor::isComputeStats).orElse(false)) {
      statsUpdater.gatherStats(ci, conf, runJobAsSelf(ci.runAs) ? ci.runAs : table.getOwner(),
          CompactorUtil.getCompactorJobQueueName(conf, ci, table), msc);
    }
```